docs/source/PaaSTA_Development.rst

```diff
@@ -0,0 +1,108 @@
+==================
+PaaSTA Development
+==================
+
+
+What is PaaSTA playground
+-------------------------
+
+PaaSTA playground allows the developer to run and debug paasta locally on an ephemeral Kubernetes cluster. Currently, it includes the following components:
+generating ``etc_paasta_playground`` and ``soa_config_playground`` directories, setup_kubernetes_job (deploy compute-infra-test-service),
+PaaSTA API, and PaaSTA CLI client.
+
+Create a Kind Cluster - K8s Cluster
+-----------------------------------
+
+Before running or debugging PaaSTA playground, you will need to create a kubernetes cluster - Kind. You can do so by running the make target
+``make create_cluster`` from ``paasta/k8s_itests`` directory.
+
+To delete the cluster, run ``make clean`` from ``paasta/k8s_itests`` directory.
+
+Debugging PaaSTA playground (in VS Code)
+----------------------------------------
+
+All components of PaaSTA playground are easy to debug in VS Code by running the debugging configurations in ``launch.json`` file. Before you start
+debugging, first run ``make vscode_settings``. This make target will add debugging settings in ``launch.json`` file.
+
+Below outlines the steps to run the debugger in VS Code and to debug PaaSTA playground components.
+
+.. _running-debugger:
+
+Running debugger (in VS Code)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to run the debugger with any of the configurations in ``launch.json``, you can go to ``Run and Debug`` tab in VS Code
+
+.. image:: run_and_debug_tab.png
+   :width: 700px
+
+and select the configuration you want to run the debugger with, then press on ``start debugging`` button
+
+.. image:: start_debugger.png
+   :width: 700px
+
+For more information on how to run the VS Code debugger, please refer to `VS Code Documentation <https://code.visualstudio.com/docs/editor/debugging>`_
+
+.. _debugging-paasta-playground:
+
+Debugging PaaSTA playground components (in VS Code)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Steps below outline running PaaSTA playground components with a debugger attached:
+
+1. Run the make target ``generate_etc_paasta_playground`` to generate the ``etc_paasta_playground`` and ``soa_config_playground`` directories and ``deployments.json`` file.
+2. [This step can be ran one time] From the ``Run and Debug`` tab in VS Code, press on ``Run setup k8s job in playground`` to deploy ``compute-infra-test-service`` service in the cluster, if it's not deployed.
+
+   Note: we're expecting to see the service in ``pending status`` as zookeeper component doesn't exist yet in paaSTA playground:
```

Comment:
do we need zk running to start pods in kind? afaik, the only thing inside our infra that uses zk in paasta is some service mesh stuff

Reply:
yes, I think so. When I run setup_kubernetes_job.py it gives me this at the end of the output:
```
Traceback (most recent call last):
  File "/nail/home/emanelsabban/pg/paasta/paasta_tools/setup_kubernetes_job.py", line 176, in setup_kube_deployments
    app.create(kube_client)
  File "/nail/home/emanelsabban/pg/paasta/paasta_tools/kubernetes/application/controller_wrappers.py", line 231, in create
    self.sync_horizontal_pod_autoscaler(kube_client)
  File "/nail/home/emanelsabban/pg/paasta/paasta_tools/kubernetes/application/controller_wrappers.py", line 308, in sync_horizontal_pod_autoscaler
    should_have_hpa = desired_hpa_spec and not autoscaling_is_paused()
  File "/nail/home/emanelsabban/pg/paasta/paasta_tools/autoscaling/autoscaling_service_lib.py", line 839, in autoscaling_is_paused
    with ZookeeperPool() as zk:
  File "/nail/home/emanelsabban/pg/paasta/paasta_tools/utils.py", line 3820, in __enter__
    hosts=load_system_paasta_config().get_zk_hosts(), read_only=True
  File "/nail/home/emanelsabban/pg/paasta/paasta_tools/utils.py", line 2084, in get_zk_hosts
    % self.directory
paasta_tools.utils.PaastaNotConfiguredError: Could not find zookeeper connection string in configuration directory: ./etc_paasta_playground/
```

It looks for a zookeeper connection string.

Reply:
ah interesting - forgot that we had this!

might be worth making a ticket for future-us to make this fail more smoothly (probably default to autoscaling enabled if we can't talk to zk?) since it doesn't seem ideal to have pending pods when there's no zk to talk to (even if this is only likely to happen when running in a playground env)

Reply:
sounds good, will create a ticket :)